scylla.go

```diff
@@ -4,17 +4,96 @@ import (
 	"fmt"
 	"math"
 	"strconv"
+	"strings"
 	"sync/atomic"
 )
 
 // scyllaSupported represents Scylla connection options as sent in SUPPORTED
 // frame.
+// FIXME: Should also follow `cqlProtocolExtension` interface.
 type scyllaSupported struct {
 	shard             int
 	nrShards          int
 	msbIgnore         uint64
 	partitioner       string
 	shardingAlgorithm string
+	lwtFlagMask       int
+}
+
+// CQL Protocol extension interface for Scylla.
+// Each extension is identified by a name and defines a way to serialize itself
+// in STARTUP message payload.
+type cqlProtocolExtension interface {
+	name() string
+	serialize() map[string]string
+}
+
+func findCQLProtoExtByName(exts []cqlProtocolExtension, name string) *cqlProtocolExtension {
+	for i := range exts {
+		if exts[i].name() == name {
+			return &exts[i]
+		}
+	}
+	return nil
+}
+
+// Top-level keys used for serialization/deserialization of CQL protocol
+// extensions in SUPPORTED/STARTUP messages.
+// Each key identifies a single extension.
+const (
+	lwtAddMetadataMarkKey = "SCYLLA_LWT_ADD_METADATA_MARK"
+)
+
+// "LWT prepared statements metadata mark" CQL Protocol Extension.
+// This extension, if enabled (properly negotiated), allows Scylla server
+// to set a special bit in prepared statements metadata, which would indicate
+// whether the statement at hand is LWT statement or not.
+//
+// This is further used to consistently choose primary replicas in a predefined
+// order for these queries, which can reduce contention over hot keys and thus
+// increase LWT performance.
+//
+// Implements cqlProtocolExtension interface.
```

Comment:
You can assert that by var _ cqlProtocolExtension = lwtAddMetadataMarkExt{}

Reply:
Can you please explain why is this needed or where do I need to put that? If we use an object that does not implement all the methods from `cqlProtocolExtension` interface in places where it's expected, then the compiler will just issue a compiler error.

Reply:
It's a pretty common pattern to also declare and not just duck type the interface implementation.

Reply:
OK, will do.